### ISSUE TITLE
No word_dict.txt in dataset

### ISSUE THREAD

**Issue by taolalong311095** [daominhkhanh20/VietnameseSpellingCorrection#1]
Please add the word_dict.txt in dataset.

**Comment by daominhkhanh20** [comment]
Sorry, This is private data in my company. You can use any open source for the Vietnamese word dictionary

**Comment by taolalong311095** [comment]
Is this the code in any papers in Vietnamese NLP, sir? Big thanks for your sharing code.  
It seems like the architecture built from this paper https://arxiv.org/pdf/2105.13578.pdf . However, I can see your finetuning in phobert (by merging PhoTweet). Your great kindness when you release this code.

**Comment by daominhkhanh20** [comment]
Thanks for your compliments. I'm also referenced from Zalo paper, SoftMasked Bert for Chinese,...
I tried experiment with  BartPho but it didn't work and the last thing I need to said that **This model is still not effective in some cases where the word is wrong but the word still makes sense.** Currently, this project needs more research for this case

**Comment by taolalong311095** [comment]
Do you have any comparison with other model such as model in Zalo paper? And your main model is still BartPho + Tweet? I do believe that if you add negative sampling or trigger the dataset as the way it wrong , your model will be more efficient. Could you tell me your time training , many epochs and how many sentences did you train?

**Comment by daominhkhanh20** [comment]
In my experience, I realize a few things:
+ BartPho isn't working because BartPho relies on BART Denoising sentence to sentence paper which input can be masked, delete,  sentence rotation,.. But in my generate data, I focus to generate sentence which has word wrong but the word make sense( I think it's the hardest part in the spelling correction)

+ In zalo paper, I think the most important in zalo model is which strategies for combining characters embedding and word embedding? Because of the limitation for GPU, we can't try many strategies(about 4 hours for 1 epoch)
+ Currently, the best model is SoftmaskedBert

We trained on about 1.5M sentences. If you researched in your university, you can try a small corpus and try for fun, but if you do for the company, I think the hardest is data. You need to generate the general data for spelling correction

Best regard

**Comment by taolalong311095** [comment]
Big thanks and respectation.

**Comment by hoangftran** [comment]
@daominhkhanh20 Hi sir, in SoftmaskedBert, how can you mapping of the tokenizer of correct sentences and error sentences? Because PhoBERT use WordPiece Tokenizer, when in SoftmaskedBert tokenize by whitespace? Thank you very much
